main.py

```diff
@@ -809,11 +858,11 @@ def _send_flow_mods(
     ):
         """Send FlowMod (and BarrierRequest) given a list of flow_dicts to switches."""
         for switch in switches:
-            for i, (flow_mod, flow, owner) in enumerate(zip(flow_mods, flows, owners)):
+            for i, (flow_mod, flow, owner) in enumerate(zip(flow_mods[switch.id], flows[switch.id], owners[switch.id])):
```

Comment:
`send_flow_mods` method needs type annotation since not you're using a dict, it'll make it clearer for future readers

Reply:
Added notations.